src/condor_utils/file_transfer.cpp

```diff
@@ -5809,13 +5828,21 @@ FileTransfer::InvokeFileTransferPlugin(CondorError &e, const char* source, const
 	if (result != TransferPluginResult::Success) {
 		std::string errorMessage;
 		std::string transferUrl;
-		plugin_stats->LookupString("TransferError", errorMessage);
+		if (!plugin_stats->LookupString("TransferError", errorMessage)) {
+			errorMessage = "File transfer plugin " + plugin +
+				" exited unexpectedly without producing an error message\n";
+		}
 		plugin_stats->LookupString("TransferUrl", transferUrl);
 		e.pushf("FILETRANSFER", 1, "non-zero exit (%i) from %s. Error: %s (%s)",
-			rc, plugin.c_str(), errorMessage.c_str(), transferUrl.c_str());
+			exit_status, plugin.c_str(), errorMessage.c_str(), transferUrl.c_str());
 		return TransferPluginResult::Error;
 	}
 
+	if (e.getFullText().empty()) {
+		e.pushf("FILETRANSFER", 1, "File transfer plugin %s exited unexpectedly "
+			"without producing an error message\n", plugin.c_str());
```

Comment:
In the case that `result == TransferPluginResult::Success`, in which case we should resonably expect `e.getFullText()` to be empty, should we really be pushing an error message?

Reply:
Thanks, this was a misplaced check that I moved somewhere else and then forgot to remove. I'll push an update.